nixos/modules/services/logging/promtail.nix

```
@@ -41,6 +41,10 @@ in {
       wantedBy = [ "multi-user.target" ];
       stopIfChanged = false;
 
+      preStart = ''
+        ${pkgs.promtail}/bin/promtail -config.file=${prettyJSON cfg.configuration} -check-syntax
```

Can't we use this instead ?
```suggestion
        ${lib.getExe pkgs.promtail} -config.file=${prettyJSON cfg.configuration} -check-syntax
```